proxy/grpcproxy/adapter/lease_client_adapter.go

```diff
@@ -39,27 +38,10 @@ func (c *ls2lc) LeaseRevoke(ctx context.Context, in *pb.LeaseRevokeRequest, opts
 }
 
 func (c *ls2lc) LeaseKeepAlive(ctx context.Context, opts ...grpc.CallOption) (pb.Lease_LeaseKeepAliveClient, error) {
-	// ch1 is buffered so server can send error on close
-	ch1, ch2 := make(chan interface{}, 1), make(chan interface{})
-	headerc, trailerc := make(chan metadata.MD, 1), make(chan metadata.MD, 1)
-
-	cctx, ccancel := context.WithCancel(ctx)
-	cli := &chanStream{recvc: ch1, sendc: ch2, ctx: cctx, cancel: ccancel}
-	lclient := &ls2lcClientStream{chanClientStream{headerc, trailerc, cli}}
-
-	sctx, scancel := context.WithCancel(ctx)
-	srv := &chanStream{recvc: ch2, sendc: ch1, ctx: sctx, cancel: scancel}
-	lserver := &ls2lcServerStream{chanServerStream{headerc, trailerc, srv, nil}}
+	cs, ss, errHandler := createChanStream(ctx)
```

Comment:
have the `createChanStream` thing setup the goroutine and totally control the error path? something like:
```go
cs := newPipeStream(ctx, func(css *chanServerStream) error {
    return c.leaseServer.LeaseKeepAlive(&ls2lcServer{ss})
})
return &ls2lcClientStream{cs}, nil
```
?